client/my-sites/backup/rewind-flow/granular-restore.tsx

```diff
@@ -162,20 +183,66 @@ const BackupGranularRestoreFlow: FunctionComponent< Props > = ( {
 		setLoading( false );
 	}, [ rewindState ] );
 
-	const onConfirm = useCallback( () => {
-		const includePaths = browserCheckList.includeList.map( ( item ) => item.id ).join( ',' );
-		const excludePaths = browserCheckList.excludeList.map( ( item ) => item.id ).join( ',' );
-		dispatch( setValidFrom( 'restore', Date.now() ) );
-		setUserHasRequestedRestore( true );
+	const preflightStatus = useSelector( ( state ) => getPreflightStatus( state, siteId ) );
+	const hasCredentials = useSelector( ( state ) => hasJetpackCredentials( state, siteId ) );
+	const credentialsAreValid = hasCredentials && ! areCredentialsInvalid;
+	const isRestoreInProgress = useSelector( ( state ) => getIsRestoreInProgress( state, siteId ) );
+	const needCredentials = useSelector( ( state ) => getDoesRewindNeedCredentials( state, siteId ) );
+	const isPreflightEnabled = config.isEnabled( 'jetpack/backup-restore-preflight-checks' );
+	const { refetch: refetchPreflightStatus } = usePreflightStatusQuery(
+		siteId,
+		// Only enable the preflight check if the user has requested a restore and we don't need credentials.
+		userHasRequestedRestore && ! needCredentials
+	);
+	const preflightCheck = useEnqueuePreflightCheck( siteId );
 
-		// Given that it may take some time for the restore to queue/start, let's add a loading state
-		// It will be removed once the restore is officially queued
-		setLoading( true );
+	useEffect( () => {
```

Comment:
We should consider creating a custom hook and reuse it on regular and granular restores. I'm taking notes so we could address this later.